drivers: video: video_sw_generator: format string mismatch causes build failure on 64-bit platforms

### Describe the bug

In `drivers/video/video_sw_generator.c` line 337, `LOG_ERR` uses `%u` for `frame_size` (`uint32_t`) while `vbuf->size` is `size_t` (`long unsigned int` on 64-bit platforms). This is caught by `-Werror` and causes a build failure on 64-bit targets such as `frdm_imx93/mimx9352/a55` (Cortex-A55).

```
drivers/video/video_sw_generator.c:337:17: error: format ‘%u’ expects argument of type ‘unsigned int’, but argument 3 has type ‘long unsigned int’ [-Werror=format=]
  337 |         LOG_ERR("Buffer too small (need %u, have %zu)", frame_size, vbuf->size);
cc1: all warnings being treated as errors
ninja: build stopped: subcommand failed.
```

### Regression

- [ x] This is a regression.

12782f1c9d7748357a79cd8dbb79b98baf5417c6 is the first bad commit
commit 12782f1c9d7748357a79cd8dbb79b98baf5417c6
Author: Kate Wang <yumeng.wang@nxp.com>
Date:   Tue Apr 28 11:33:48 2026 +0800

    drivers: video: sw_generator: Add JPEG and PNG format support
    
    Add support for JPEG and PNG compressed formats to the video software
    generator driver. This enables testing of compressed video formats
    without requiring actual camera hardware.
    
    The compressed frame buffers are included at compile time via
    generated header files, with configurable file names to support
    custom test patterns.
    
    Signed-off-by: Kate Wang <yumeng.wang@nxp.com>

### Steps to reproduce

```
west build -p -b frdm_imx93/mimx9352/a55 samples/subsys/usb/uvc
```

Or via twister:
```
scripts/twister -p frdm_imx93/mimx9352/a55 -T samples/subsys/usb/uvc --build-only
```

**Zephyr version:** `v4.4.0-3337-g07e34e68d340`

Also reproduced in NXP internal CI (`bitbucket_build_platform` build #9667).

### Root cause

`frame_size` is `uint32_t` but the format specifier `%u` is treated as `unsigned int`. On 64-bit platforms (Cortex-A55), `size_t` is `long unsigned int`, causing the compiler to flag the mismatch as an error via `-Werror=format=`.

### Suggested fix

Use `PRIu32` for portable `uint32_t` formatting:

```c
// drivers/video/video_sw_generator.c:337
// Before:
LOG_ERR("Buffer too small (need %u, have %zu)", frame_size, vbuf->size);

// After:
LOG_ERR("Buffer too small (need %" PRIu32 ", have %zu)", frame_size, vbuf->size);
```

### Related

- PR #85968 (`drivers: video: sw_generator: add more formats and try to simplify`) — last PR to touch `video_sw_generator.c`
- PR #92370 (`Video: Rework buffer management`) — open draft reworking buffer management types

--- a/drivers/video/video_sw_generator.c
+++ b/drivers/video/video_sw_generator.c
@@ -13,6 +13,7 @@
 #include <zephyr/logging/log.h>
 #include <zephyr/sys/byteorder.h>
 #include <zephyr/sys/util.h>
+#include <inttypes.h>
 
 #include "video_ctrls.h"
 #include "video_device.h"
@@ -334,7 +335,8 @@ static int video_sw_generator_fill_compressed(struct video_buffer *vbuf, bool hf
 
 	/* Check if buffer is large enough */
 	if (vbuf->size < frame_size) {
-		LOG_ERR("Buffer too small (need %u, have %zu)", frame_size, vbuf->size);
+		LOG_ERR("Buffer too small (need %" PRIu32 ", have %" PRIu32 ")", frame_size,
+			vbuf->size);
 		return -ENOMEM;
 	}
 

